src/components/external-dns/index.tsx

```diff
@@ -0,0 +1,184 @@
+  messages: Array<string>;
+}> = ({ status, messages }) => (
+  <Alert {...AlertTemplates[status]}>
+    <div className="grid grid--gap-medium">
+      {messages.map((msg, i) => (
+        <Typography key={i}>{msg}</Typography>
+      ))}
+    </div>
+  </Alert>
+);
+
+const CertificateExpiry: FunctionComponent<{ expires: string }> = ({
+  expires,
+}) => {
+  const expiresIn = differenceInDays(new Date(expires), new Date());
+  return <>{dayPluraliser(expiresIn)}</>;
+};
+
+export const ExternalDNSList: FunctionComponent<{
+  externalDnsList: Array<ExternalDns>;
+  fqdnElem?: (externalDns: ExternalDns) => React.JSX.Element;
+}> = ({ externalDnsList: externalDNSList, fqdnElem }) => {
+  const sortedExternalDNSList = useGetSortedExternalDNS(externalDNSList);
+  const [expandedRows, setExpandedRows] = useState<Record<string, boolean>>({});
+  const expandRow = useCallback<(name: string) => void>(
+    (name) => setExpandedRows((x) => ({ ...x, [name]: !x[name] })),
+    []
+  );
+
+  return (
+    <Table>
+      <Table.Head>
+        <Table.Row>
+          <Table.Cell width={40} />
+          <Table.Cell>FQDN</Table.Cell>
+          <Table.Cell width={150}>Expires</Table.Cell>
+          <Table.Cell width={150}>Status</Table.Cell>
+          <Table.Cell width={150}>TLS Automation</Table.Cell>
+        </Table.Row>
+      </Table.Head>
+      <Table.Body>
+        {sortedExternalDNSList
+          ?.map((v) => ({
+            externalDNS: v,
+            hasCertificates: v.tls.certificates?.length > 0,
+            certificateExpiry:
+              v.tls.certificates?.length > 0
+                ? v.tls.certificates[0].notAfter
+                : null,
+            hasMessages: v.tls.statusMessages?.length > 0,
+            expanded: !!expandedRows[v.fqdn],
+          }))
+          .map(
+            ({
+              externalDNS,
+              hasCertificates,
+              certificateExpiry,
+              hasMessages,
+              expanded,
+            }) => (
+              <Fragment key={externalDNS.fqdn}>
+                <Table.Row
+                  className={clsx({
+                    'border-bottom-transparent': expanded,
+                  })}
+                >
+                  <Table.Cell className="fitwidth padding-right-0">
+                    {(hasCertificates || hasMessages) && (
+                      <Typography
+                        link
+                        as="span"
+                        onClick={() => expandRow(externalDNS.fqdn)}
+                      >
+                        <Icon
+                          data={expanded ? chevron_up : chevron_down}
+                          role="button"
+                          title="Toggle more information"
+                        />
+                      </Typography>
+                    )}
+                  </Table.Cell>
+                  <Table.Cell>
+                    {fqdnElem ? (
+                      fqdnElem(externalDNS)
```

Comment:
Consider inlining it, or use he `children`prop?


I prefer being explicit when defining the fields instead of `{...{some:properties}}`: 
```jsx
<ExternalDNSFQDNField
  externalDns={externalDNS}
  appName={appName}
  envName={envName}
  componentName={compnentName}
/>
```

By using children you could do this in `external-dns.txt`:
```jsx
<ExternalDNSList
  externalDnsList={externalDNSList}
>
   {list.length === 0 
     ? <Typography>{externalDNS.fqdn}</Typography>
     : list.map(externalDNS => (
      <ExternalDNSFQDNField
        externalDns={externalDNS}
        appName={appName}
        envName={envName}
        componentName={compnentName}
      />
    ))
  }
</ExternalDNSList>
```

And then in `ExternalDNSList` add `children` in the props list (`ReactNode`) and use it like this:
```
<Table.Cell>
{children}
</Table.Cell>
````

Reply:
fixed the first comment about explicit props

Reply:
refactored external dns list